### PR TITLE
Rename microsoft-reactnative-sampleapps to sample-apps

### PR DESCRIPTION
**Pull request by NickGerleman**
This is part of a series of PRs cleaning up naming in the `packages directory for consistency`. Certain conventions have grown over time, but not all packages follow them. This should create clarity on informal convnetions on where packages go and how to name them.

The changes across PRs
- `react-native-win32` and `react-native-win32-tester` to `@office-iss`
- `react-native-windows-codegen` to `@react-native-windows/codegen`
- `microsoft-reactnative-sampleapps to sample-apps`
- `E2ETest` to `e2e-test-app`
- `IntegrationTest` to `integration-test-app`

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/7003)

### CODE REVIEW

**Review by dannyvv — APPROVED** [review]

File: packages/sample-apps/windows/SampleAppCPP/SampleAppCpp.vcxproj [hunk 1]
```diff
@@ -14,7 +14,7 @@
     <ApplicationTypeRevision>10.0</ApplicationTypeRevision>
     <WindowsTargetPlatformVersion>10.0.18362.0</WindowsTargetPlatformVersion>
     <WindowsTargetPlatformMinVersion>10.0.16299.0</WindowsTargetPlatformMinVersion>
-    <AppxPackageSigningEnabled>True</AppxPackageSigningEnabled>
+    <AppxPackageSigningEnabled>False</AppxPackageSigningEnabled>
```
Comment (asklar) [inline packages/sample-apps/windows/SampleAppCPP/SampleAppCpp.vcxproj:17]: Why this change?
Same with three other vcxproj change

Comment (vmoroz) [reply packages/sample-apps/windows/SampleAppCPP/SampleAppCpp.vcxproj:17]: Usually removal of the keys causes the UWP apps not run. It would be nice not to do the signing related changes.

**Comment by vmoroz** [comment]
@NickGerleman  please make sure that the sample app are running after this change. Removal of signing keys in UWP apps usually makes them not runnable.

File: packages/sample-apps/windows/SampleAppCPP/SampleAppCpp.vcxproj (hunk 1 above)
Comment (asklar) [reply packages/sample-apps/windows/SampleAppCPP/SampleAppCpp.vcxproj:17]: That is what's breaking the CI:
```
Error: The package or bundle is not digitally signed or its signature is corrupted.
× Installing new version of the app
- Failed to deploy: Installing new version of the app
```

Comment (NickGerleman) [reply packages/sample-apps/windows/SampleAppCPP/SampleAppCpp.vcxproj:17]: Yes, I can revert it. This started with keys being removed bc they weren't exempted from the gitignore, but we also have some apps with signing disabled, and was curious if that would break anything.

Wondering if we've only ever tested the other apps with deploy from layout, or why they wouldn't see issues?

**Comment by asklar** [comment]
@NickGerleman FYI that this broke the Docs CI since we have links to  https://github.com/microsoft/react-native-windows/tree/master/packages/microsoft-reactnative-sampleapps
See CI run here: https://github.com/microsoft/react-native-windows-samples/pull/377/checks?check_run_id=1850217038
